sdk/go/common/workspace/plugins.go

```diff
@@ -1806,19 +1806,34 @@ func getPluginInfoAndPath(
 						(stat.Mode()&0o100 != 0 || runtime.GOOS == windowsGOOS) {
 						logging.V(6).Infof("GetPluginPath(%s, %s, %v): found next to current executable %s",
 							kind, name, version, candidate)
-
-						return &PluginInfo{
-							Kind: kind,
-							Name: name,
-							Path: filepath.Dir(candidate),
-						}, candidate, nil
+						bundledPath = candidate
+						break
 					}
 				}
 			}
 		}
 	}
 
-	// Otherwise, check the plugin cache.
+	// We prefer the ambient path, but we need to check if this is the same as the bundled
+	// path to decide if we're warning or not.
+	if ambientPath != "" {
+		if ambientPath != bundledPath {
+			d.Warningf(diag.Message("", "using %s from $PATH at %s"), filename, ambientPath)
+		}
+		return &PluginInfo{
+			Kind: kind,
+			Name: name,
+			Path: filepath.Dir(ambientPath),
+		}, ambientPath, nil
+	} else if bundledPath != "" {
+		return &PluginInfo{
+			Kind: kind,
+			Name: name,
+			Path: filepath.Dir(bundledPath),
+		}, bundledPath, nil
```

Comment:
minor: I think this would be cleaner if we didn't have the two separate returns. We could have a block calculating a plugin path, and then just building PluginInfo from that.

```go
pluginPath := bundledPath
if ambientPath != "" {
	if ambientPath != pluginPath {
		// warn
	}
	pluginPath = ambientPath
}
if pluginPath != "" {
	return &PluginInfo{..}, pluginPath, nil
}
```